src/components/calculator.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import '../stylesheet/Calculator.css';
+import calculate from '../logic/calculate';
+
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.clickHandler = this.clickHandler.bind(this);
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+  }
+
+  clickHandler(e) {
+    this.setState((state) => calculate(state, e.target.textContent));
+  }
```

Comment:
- [ ] Well done @obibaadoma you did a really great job by implementing the things asked by the previous code reviewer but it seems like you missed the point mentioned by your previous code reviewer about using the `useState` hook. It seems like you are using `class-based` components and you are using the `this.state` or `setState` for managing the state of your component. Kindly adhere to the project requirement according to which you should use `useState` and `props` to manage the state and pass the data from one component to another component, and the `useState` hook is only used when you have the `functional` component, which means you need to refactor your code and transform your `class-based` component to `function` based component. Please follow the instructions and use the functional components because functional components approach makes our code more maintainable, easy to read and even reduce the number of lines of code. On the other handside the new hooks are only accessible in the function-based component, So sooner or later you would switch to functional styling approach and I think it is the best time to do. 

### Requirement
![image](https://github.com/obibaadoma/math-magic/assets/66476329/8d12c833-e998-4a3c-8033-2b424c51a642)

Reply:
will consider that. thanks